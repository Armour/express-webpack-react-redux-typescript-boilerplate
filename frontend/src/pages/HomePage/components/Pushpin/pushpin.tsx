import React from 'react';
import { Translation } from 'react-i18next';

const styles = require('./pushpin.scss');

interface IPushpinProps {
  color: string;
  depth: string;
}

export class Pushpin extends React.Component<IPushpinProps> {
  public render() {
    return (
      <Translation ns='homePage'>
        {(t) => (
          <div id={this.props.color} className={`${styles.pushpin} ${this.props.color} ${this.props.depth}`}>
            <nav className='pushpin-nav pin-top' data-target={'pushpin-' + this.props.color}>
              <div className={`nav-wrapper ${this.props.color} pushpin-${this.props.color}`}>
                <div className='container'>
                  <a href='#' className='brand-logo'>{t('pushpin.' + this.props.color)}</a>
                  <ul id={`nav-demo-${this.props.color}`} className='right hide-on-med-and-down'>
                    <li><a href='#'>{t('pushpin.linkOne')}</a></li>
                    <li><a href='#'>{t('pushpin.linkTwo')}</a></li>
                  </ul>
                </div>
              </div>
            </nav>
          </div>
        )}
      </Translation>
    );
  }
}

export default Pushpin;
