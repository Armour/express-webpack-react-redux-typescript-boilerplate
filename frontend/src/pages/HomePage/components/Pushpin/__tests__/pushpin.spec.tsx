import React from 'react';
import TestRenderer from 'react-test-renderer';

import { Pushpin } from '../pushpin';

describe('Pushpin', () => {
  it('should renders correctly', () => {
    const renderer = TestRenderer.create(
      <Pushpin color='red' depth='base' />,
    );
    expect(renderer).toMatchSnapshot();
    renderer.unmount();
  });
});
