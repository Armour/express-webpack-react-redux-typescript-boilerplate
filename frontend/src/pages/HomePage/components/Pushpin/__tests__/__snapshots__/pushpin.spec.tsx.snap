// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pushpin should renders correctly 1`] = `
<div
  className="pushpin red base"
  id="red"
>
  <nav
    className="pushpin-nav pin-top"
    data-target="pushpin-red"
  >
    <div
      className="nav-wrapper red pushpin-red"
    >
      <div
        className="container"
      >
        <a
          className="brand-logo"
          href="#"
        >
          pushpin.red
        </a>
        <ul
          className="right hide-on-med-and-down"
          id="nav-demo-red"
        >
          <li>
            <a
              href="#"
            >
              pushpin.linkOne
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              pushpin.linkTwo
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
